File: src/Model/Cell.cpp
#include "Cell.h"

Cell::Cell(Coords coords, double cellLength)
	: m_coords(coords)
	, m_cellLength(cellLength)
	, m_cost(1.0)
{
}

void Cell::AddConnection(Cell* neighbor)
{
	m_connections.push_back(neighbor);
}

void Cell::SetCost(double cost)
{
	m_cost = cost;
}

Coords Cell::GetCoords() const
{
	return m_coords;
}

const std::vector<Cell*>& Cell::GetConnections() const
{
	return m_connections;
}

double Cell::GetCost() const
{
	return m_cost;
}

double Cell::GetCellLength() const
{
	return m_cellLength;
}

File: src/Model/Cell.ts
import {Coords} from './Coords';

export class Cell {
    private readonly coords: Coords;
    private readonly cellLength: number;
    private cost: number;
    private readonly connections: Cell[];

    constructor(coords: Coords, cellLength: number) {
        this.coords = coords;
        this.cellLength = cellLength;
        this.cost = 1.0;
        this.connections = [];
    }

    addConnection(neighbor: Cell) {
        this.connections.push(neighbor);
    }

    setCost(cost: number) {
        this.cost = cost;
    }

    getCoords(): Coords {
        return this.coords;
    }

    getConnections(): Cell[] {
        return this.connections;
    }

    getCost(): number {
        return this.cost;
    }

    getCellLength(): number {
        return this.cellLength;
    }
}
